test/CompanyService.Business.UnitTests/Commands/Department/CreateDepartmentCommandTests.cs

```diff
@@ -77,108 +66,186 @@ public void OneTimeSetUp()
+
+            _autoMock
+                .Setup<IDepartmentRepository, bool>(x => x.DoesNameExist(_request.Name))
+                .Returns(false);
         }
 
         [Test]
         public void ShouldThrowForbiddenExceptionWhenUserIsNotAdminAndNotEnoughRights()
         {
-            _accessValidatorMock
-                .Setup(x => x.IsAdmin(null))
+            _autoMock
+                .Setup<IAccessValidator, bool>(x => x.IsAdmin(null))
                 .Returns(false);
 
-            _accessValidatorMock
-                .Setup(x => x.HasRights(Rights.AddEditRemoveDepartments))
+            _autoMock
+                .Setup<IAccessValidator, bool>(x => x.HasRights(Rights.AddEditRemoveDepartments))
                 .Returns(false);
 
-            Assert.Throws<ForbiddenException>(() => _command.Execute(_request));
-            _companyRepositoryMock.Verify(x => x.Get(It.IsAny<GetCompanyFilter>()), Times.Never);
-            _repositoryMock.Verify(x => x.CreateDepartment(It.IsAny<DbDepartment>()), Times.Never);
+            var ex = Assert.Throws<ForbiddenException>(() => _command.Execute(_request));
+            Assert.That(ex.Message, Is.EqualTo("Not enough rights."));
+
+            _autoMock.Verify<ICompanyRepository, DbCompany>(
+                x => x.Get(It.IsAny<GetCompanyFilter>()), 
+                Times.Never);
+
+            _autoMock.Verify<IDepartmentRepository, Guid>(
+                x => x.CreateDepartment(It.IsAny<DbDepartment>()), 
+                Times.Never);
         }
 
-        /*[Test]
-        public void ShouldThrowArgumentNullExceptionWhenDepartmentRequestIsNull()
+        [Test]
+        public void ShouldThrowValidationException()
         {
+            _autoMock
+                .Setup<ICreateDepartmentRequestValidator, bool>
+                (x => x.Validate(It.IsAny<IValidationContext>()).IsValid)
+                .Returns(false);
 
-            _companyRepositoryMock
-                .Setup(x => x.Get(null))
-                .Returns(new DbCompany { Id = _companyId });
+            Assert.Throws<ValidationException>(() => _command.Execute(_request));
 
-            _mapperMock
-                .Setup(x => x.Map(null, _companyId))
-                .Throws(new ArgumentNullException());
+            _autoMock.Verify<ICompanyRepository, DbCompany>(
+                x => x.Get(It.IsAny<GetCompanyFilter>()), 
+                Times.Never);
 
-            Assert.Throws<ArgumentNullException>(() => _command.Execute(null));
-            _companyRepositoryMock.Verify(x => x.Get(null), Times.Once);
-            _repositoryMock.Verify(x => x.CreateDepartment(_dbDepartment), Times.Never);
-        }*/
+            _autoMock.Verify<IDepartmentRepository, Guid>(
+                x => x.CreateDepartment(It.IsAny<DbDepartment>()), 
+                Times.Never);
+        }
 
         [Test]
-        public void ShouldAddNewDepartmentSuccessfully()
+        public void CompanyDoesNotExistTest()
```

Comment:
rename

Reply:
should throw exc when company doesn't exist

Reply:
Done